tmt/utils.py

```diff
@@ -4727,17 +4727,21 @@ def validate_git_status(test: 'tmt.base.Test') -> tuple[bool, str]:
 
 def generate_runs(
         path: Path,
-        id_: Optional[str] = None) -> Iterator[Path]:
+        id_: tuple[str, ...]) -> Iterator[Path]:
     """ Generate absolute paths to runs from path """
     # Prepare absolute workdir path if --id was used
-    if id_:
-        run_path = Path(id_)
-        if '/' not in id_:
-            run_path = path / run_path
-        if run_path.is_absolute():
-            if run_path.exists():
+    run_path = None
+    for id_name in id_:
+        if id_name:
+            run_path = Path(id_name)
+            if '/' not in id_name:
+                run_path = path / run_path
+            if run_path.is_absolute() and run_path.exists():
                 yield run_path
-            return
+        else:
+            raise tmt.utils.GeneralError("id can not be empty.")
```

Comment:
The message could be a bit more helpful :) The world is full of various "id"s, let's give the user more context, it wouldn't cost us much. How about something like "Value of '--id' option cannot be an empty string."? We want to help the user to find the mistake quickly, and we know the value comes from the `--id` option, and "empty string" is more specific than "empty".